docs/content/docs/recipes/audience-react.md

```diff
@@ -0,0 +1,396 @@
+
+```js
+  import { useState, useCallback } from "react";
+  import { AudienceDisplay } from "./AudienceDisplay";
+  import { SelectUser } from "./SelectUser"
+
+  export const App = () => {
+    // TODO 1: Define state variables to handle view changes and user input
+    return (
+    // TODO 2: Return view components
+    );
+  }
+```
+
+1. Replace `TODO 1` with the following code. Note that the values `userId` and `containerId` will come from a `SelectUser` component through the `handleSelectUser` function. The `displayAudience` variable is used to switch between the member list view and the userId selection view. In the case that a user inputs an invalid `audienceId` they will be redirected to the userId selection view by the `handleContainerNotFound` function.
+
+```js
+  const [displayAudience, setDisplayAudience] = useState(false);
+  const [userId, setUserId] = useState();
+  const [containerId, setContainerId] = useState();
+
+  const handleSelectUser = useCallback((userId, containerId) => {
+    setDisplayAudience(true)
+    setUserId(userId);
+    setContainerId(containerId);
+  }, [displayAudience, userId, containerId]);
+
+  const handleContainerNotFound = useCallback(() => {
+    setDisplayAudience(false)
+  }, [setDisplayAudience]);
+```
+
+1. Replace `TODO 2` with the following code. As stated above, the `displayAudience` variable will switch between the `AudienceDisplay` component and `SelectUser` component. Also, functions to update the state variables are passed into components as properties.
+
+```js
+  (displayAudience) ?
+  <AudienceDisplay userId={userId} containerId={containerId} onContainerNotFound={handleContainerNotFound}/> :
+  <SelectUser onSelectUser={handleSelectUser}/>
+```
+
+### Set up AudienceDisplay component
+
+1. Create and open a file `\src\AudienceDisplay.js` in your code editor. Add the following `import` statements:
+
+```js
+  import { useEffect, useState } from "react";
+  import { SharedMap } from "fluid-framework";
+  import { AzureClient } from "@fluidframework/azure-client";
+  import { InsecureTokenProvider } from "@fluidframework/test-client-utils"
+```
+
+1. Add the following functional components and helper functions:
+
+```js
+  const tryGetAudienceObject = async (userId, userName, containerId) => {
+    // TODO 1: Configure the client and container
+    // TODO 2: Configure the container
+    // TODO 3: Get the container and services from the Fluid service.
+    // TODO 4: Return the Audience object.
+  }
+
+  export const AudienceDisplay = (props) => {
+    //TODO 5: Configure user id, user name, and state variables
+    //TODO 6: Set state variables and set event listener on component mount
+    //TODO 7: Return list view
+  }
+
+  const AudienceList = (data) => {
+    //TODO 8: Append view elements to list array for each member
+    //TODO 9: Return list of member elements
+  }
```

Comment:
Why do these functions have noun names instead of verb names as functions normally do? And why do they begin with a capital letter? Shouldn't these be named displayAudience and getAudience?

Reply:
These functions are React function components that return React elements. Since they are alternatives to class components, they tend to follow the same naming conventions as classes.

Reply:
> These functions are React function components that return React elements. Since they are alternatives to class components, they tend to follow the same naming conventions as classes.

Got it. thanks.